src/OneTimeTokenProvider.js

```diff
@@ -0,0 +1,43 @@
+import Connection from './Connection'
+
+export default class OneTimeTokenProvider {
+  constructor(publicAPIKey, options) {
+    this.publicAPIKey = publicAPIKey
+    this.options = options
+    this.token = null
+    this.timeout = null
+    this.marginToExpire = 0.05 // a margin to prevent unexpected expiration (5% of the time)
+  }
+
+  get() { return this.retrieve().then(() => this.getAndClearToken()) }
+
+  retrieve() {
+    if (this.token != null) { return Promise.resolve() }
+    return this.issue().then(data => this.storeToken(data))
+  }
+
+  issue() {
```

Comment:
I had an idea to make the naming more distinct (compared to retrieve and get). How about renaming `issue()` to `requestToken()`, as it generates request to the server (to get a token)? And I think the server issues the token.. not the client.
What are your thoughts on this?

Reply:
I think issuing tokens by this class is not wrong.
According to OOP, the implementation details should be hidden. When we follow this idea, a caller should not know the implementation details.
In this case, `OneTimeTokenProvider` is the class that issues one-time tokens in an unknown way.
Maybe, this can be said as a delegation.
This is the reason why I think we can use the name `issue` for this method.
What do you feel about this thought?

Reply:
Your point is very good. Implementation details should be hidden. And from the perspective of an unknowing caller, it looks like `OneTimeTokenProvider` issues a new token. I wasn't sure that this function will be used by a caller directly (to get a token without timeout or cache handling).